### DISCUSSION TITLE
[Idea] Replace jQuery AJAX by Fetch API for native modules

### DISCUSSION THREAD

**Discussion by leemyongpakvn** [PrestaShop/PrestaShop#34239] [Ideas]
### The problem

At least these 3 native modules are using JQuery AJAX.
1. [productomments](https://github.com/PrestaShop/productcomments/blob/v6.0.2/views/js/list-comments.js#L184)
2. [ps_searchbar](https://github.com/PrestaShop/ps_searchbar/blob/v2.1.3/ps_searchbar.js#L35)
3. [ps_shoppingcart](https://github.com/PrestaShop/ps_shoppingcart/blob/v2.0.7/ps_shoppingcart.js#L65)
4. [blockreassurance](https://github.com/PrestaShop/blockreassurance/blob/v5.1.4/_dev/back/index.js#L39-L40)

### The solution

We need to modernize them by using Fetch API.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by matks** [comment]
Hi, I think it's a good idea but for the record: what are the advantages of FETCH against jQuery AJAX? (it's always good to motivate such a change with the background context, like ADR)

**Reply by boherm** [reply]
Sometime, some dev install jQuery (lite or full) just to have ajax function to retrieve some informations...
Same with `axios` library, it's better to use fetch to avoid install this kind of libraries only for this behaviour.

So yeah, I think that could be a good idea to switch progressively to fetch function.

(FYI: https://caniuse.com/fetch)

**Comment by leemyongpakvn** [comment]
@boherm Thanks for the answer and the reference link :)
@matks There are more details at https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch. For short, we need to replace old standard by new standard, like the way we chose to migrate from Legacy PS codebase to Symfony.
